### DISCUSSION TITLE
docs: Include generated ER diagram

### DISCUSSION THREAD

**Discussion by marcelstoer** [chatwoot/chatwoot#7998] [Q&A]
### Is your feature or enhancement related to a problem? Please describe.

I would like https://www.chatwoot.com/docs/product to contain a (generated) ER diagram of the Chatwoot RDBMS schema.

### Describe the solution you'd like

Run `rails db:setup` or similar against an SQLite or H2 database in a release build. Then run SchemaSpy against that database and include the generated ER diagram (SVG) in the documentation for that release.

### Describe alternatives you've considered

There's https://drawsql.app/templates/chatwoot but since it was last updated 3y ago it's likely hopelessly outdated.

### Additional context

Why are foreign keys in the Chatwoot schema largely absent? I generated a one-off ER diagram which is...marginally useful as such. It's useful in that in conveys that the schema is dozens of orphan tables.

![chatwoot_dev - public](https://github.com/chatwoot/chatwoot/assets/624195/d5eb9bc2-5296-4554-bd3d-2278051683d2)

P.S. Only the not-yet-released https://github.com/schemaspy/schemaspy/pull/1251 will allow including orphan tables in the SchemaSpy ER diagram. The above was, therefore, generated with a different tool.

**Comment by pranavrajs** [comment]
Moving this to discussions as it is not bug or a feature request in the product. 

@marcelstoer I've not used this, however you can use something like https://github.com/voormedia/rails-erd or https://dbdiagram.io/ to generate the ER diagram based on the schema.rb

**Reply by marcelstoer** [reply]
I treat product documentation as a feature 😄 Hence, I created it in that group.

Thanks for the feedback and the hints to those two services.

**Reply by marcelstoer** [reply]
Can you maybe also comment on the foreign key issue?
